### PR TITLE
Prepare for Release 3.4.0 

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml
Comment (Madhu-1) [inline deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml]: this comment is not valid?

**Review by Madhu-1 — COMMENTED** [review]
Missing upgrade doc?

File: deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml
Comment (humblec) [reply deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml]: @Madhu-1 thanks... corrected it..

**Comment by humblec** [comment]
> Missing upgrade doc?

@pkalever will be pushing the upgrade doc, once thats in only we will merge this. :+1:

File: charts/ceph-csi-cephfs/Chart.yaml
Comment (nixpanic) [inline charts/ceph-csi-cephfs/Chart.yaml:7]: this still includes `-canary`?

File: charts/ceph-csi-rbd/Chart.yaml
Comment (nixpanic) [inline charts/ceph-csi-rbd/Chart.yaml:7]: here  too

File: charts/ceph-csi-cephfs/Chart.yaml
Comment (humblec) [reply charts/ceph-csi-cephfs/Chart.yaml:7]: @nixpanic yeah, we keep this https://github.com/ceph/ceph-csi/pull/2046/files#diff-f1b11c5f81f992e01a7019d45b7fc930ca4f6268b1e7ff018ee5a327950b4e2d

File: charts/ceph-csi-rbd/Chart.yaml
Comment (humblec) [reply charts/ceph-csi-rbd/Chart.yaml:7]: @nixpanic  https://github.com/ceph/ceph-csi/pull/2046/files#diff-f1b11c5f81f992e01a7019d45b7fc930ca4f6268b1e7ff018ee5a327950b4e2d

**Review by Madhu-1 — APPROVED** [review]

**Comment by humblec** [comment]
@nixpanic can you please revisit this PR with priority?

**Comment by humblec** [comment]
@Madhu-1 can we add `ready-to-merge` as this block the release?

**Comment by humblec** [comment]
@Madhu-1 this is interesting , while I do `retest` the tests are not restarting. not sure again why this is happening :thinking:

**Comment by Madhu-1** [comment]
> @Madhu-1 this is interesting , while I do `retest` the tests are not restarting. not sure again why this is happening

Not sure what is the problem. may be an extra space?

**Comment by Madhu-1** [comment]
> /retest ci/centos/mini-e2e/k8s-1.19

```
Jul 29 07:47:35.159: INFO: will run kubectl (create) again (21 seconds elapsed)
Jul 29 07:47:35.159: FAIL: failed to create vault psp failed to run kubectl: error running /usr/bin/kubectl --server=https://192.168.39.225:8443 --kubeconfig=/root/.kube/config --namespace=cephcsi-e2e-806a3c06 create -f -:
Command stdout:

stderr:
Error from server (AlreadyExists): error when creating "STDIN": podsecuritypolicies.policy "rbd-csi-vault-token-review-psp" already exists
Error from server (AlreadyExists): error when creating "STDIN": roles.rbac.authorization.k8s.io "rbd-csi-vault-token-review-psp" already exists
Error from server (AlreadyExists): error when creating "STDIN": rolebindings.rbac.authorization.k8s.io "rbd-csi-vault-token-review-psp" already exists
```

**Comment by humblec** [comment]
> > @Madhu-1 this is interesting , while I do `retest` the tests are not restarting. not sure again why this is happening
> 
> Not sure what is the problem. may be an extra space?

Not sure either, will watch next time . Thanks for retriggering.

**Comment by Madhu-1** [comment]
> /test ci/centos/mini-e2e/k8s-1.21

`X Exiting due to GUEST_PROVISION: Failed to start host: creating host: create: Error creating machine: Error in driver during machine creation: IP not available after waiting: machine minikube didn't return IP after 1 minute
`

**Comment by Madhu-1** [comment]
@humblec can you please send a PR to revert back changes and to push release canary image and charts

**Pull request merged**
